esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2092,52 +2055,75 @@ void WsWuInfo::getWorkunitThorSlaveLog(const char *groupName, const char *ipAddr
     OwnedIFileIOStream ios = createBufferedIOStream(rIO);
     if (slaveNum > 0)
     {
-        StringBuffer line;
-        bool eof = false;
-        bool include = false;
+        readWorkunitLog(ios.get(), buf, outFile);
+    }
+    else
+    {//legacy wuid
+        appendIOStreamContent(buf, ios.get(), forDownload);
+    }
+}
 
-        VStringBuffer startwuid("Started wuid=%s", wuid.str());
-        VStringBuffer endwuid("Finished wuid=%s", wuid.str());
+void WsWuInfo::readWorkunitLog(OwnedIFileIOStream ios, MemoryBuffer& buf, const char* outFile)
+{
+    VStringBuffer startwuid("Started wuid=%s", wuid.str());
+    VStringBuffer endwuid("Finished wuid=%s", wuid.str());
 
-        const char *sw = startwuid.str();
-        const char *ew = endwuid.str();
+    const char *sw = startwuid.str();
+    const char *ew = endwuid.str();
```

Comment:
trivial: don't need .str(); cast operator will.
Not sure it's worth assigning to sw/eq either, it's only 1 virtual call per line (the ios->read per character is far more of an issue).

Reply:
Will fix.